services/app-web/src/pages/StateSubmission/RateDetails/V2/RateDetailsV2.tsx

```diff
@@ -196,8 +199,17 @@ const RateDetailsV2 = ({
     const initialRequestLoading = fetchContractLoading || fetchRateLoading
     const initialRequestError = fetchContractError || fetchRateError
     const previousDocuments: string[] = []
+    const submissionIDForHeading =
+        fetchContractData?.fetchContract.contract?.draftRevision
+            ?.contractName ||
+        fetchRateData?.fetchRate.rate?.draftRevision?.formData
+            .rateCertificationName
+
+    useEffect(() => {
+        updateHeading({ customHeading: submissionIDForHeading ?? undefined })
```

Comment:
FYI calling `updateHeading` is how I fixed the header issue @pearl-truss with the v2 pages. THe old pages did this in `useHealthPlanPackageForm`. 

 I can fix on review and submit v2 also once your work lands.

Reply:
@haworku thanks! my work just landing and I'm reviewing this PR now